plugins/module_utils/sdkmanager.py

```diff
@@ -0,0 +1,161 @@
+    return __channel_map[channel_name]
+
+
+def sdkmanager_runner(module, **kwargs):
+    return CmdRunner(
+        module,
+        command='sdkmanager',
+        arg_formats=dict(
+            state=cmd_runner_fmt.as_map(__state_map),
+            name=cmd_runner_fmt.as_list(),
+            installed=cmd_runner_fmt.as_fixed("--list_installed"),
+            list=cmd_runner_fmt.as_fixed('--list'),
+            newer=cmd_runner_fmt.as_fixed("--newer"),
+            sdk_root=cmd_runner_fmt.as_opt_eq_val("--sdk_root"),
+            channel=cmd_runner_fmt.as_func(lambda x: ["{0}={1}".format("--channel", __map_channel(x))])
+        ),
+        force_lang="auto",  # Without this, sdkmanager binary crashes
+        **kwargs
+    )
+
+
+class Package:
+    def __init__(self, name):
+        self.name = name
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __ne__(self, other):
+        if not isinstance(other, Package):
+            return True
+        return self.name != other.name
+
+    def __eq__(self, other):
+        if not isinstance(other, Package):
+            return False
+
+        return self.name == other.name
+
+
+class SdkManagerException(Exception):
+    pass
+
+
+class AndroidSdkManager(object):
+    _RE_INSTALLED_PACKAGES_HEADER = re.compile(r'^Installed packages:$')
+    _RE_UPDATABLE_PACKAGES_HEADER = re.compile(r'^Available Updates:$')
+
+    # Example: '  platform-tools     | 27.0.0  | Android SDK Platform-Tools 27 | platform-tools  '
+    _RE_INSTALLED_PACKAGE = (
+        re.compile(r'^\s*(?P<name>\S+)\s*\|\s*\S+\s*\|\s*.+\s*\|\s*(\S+)\s*$')
+    )
+
+    # Example: '   platform-tools | 27.0.0    | 35.0.2'
+    _RE_UPDATABLE_PACKAGE = re.compile(r'^\s*(?P<name>\S+)\s*\|\s*\S+\s*\|\s*\S+\s*$')
+
+    _RE_UNKNOWN_PACKAGE = re.compile(r'^Warning: Failed to find package \'(?P<package>\S+)\'\s*$')
+    _RE_ACCEPT_LICENSE = re.compile(r'^The following packages can not be installed since their licenses or those of '
+                                    r'the packages they depend on were not accepted')
+
+    def __init__(self, runner):
+        self.runner = runner
+
+    def get_installed_packages(self):
+        with self.runner('installed sdk_root channel') as ctx:
+            rc, stdout, stderr = ctx.run()
+            return self._parse_packages(stdout, self._RE_INSTALLED_PACKAGES_HEADER, self._RE_INSTALLED_PACKAGE)
+
+    def get_updatable_packages(self):
+        with self.runner('list newer sdk_root channel') as ctx:
+            rc, stdout, stderr = ctx.run()
+            return self._parse_packages(stdout, self._RE_UPDATABLE_PACKAGES_HEADER, self._RE_UPDATABLE_PACKAGE)
+
+    def apply_packages_changes(self, packages):
+        """ Install or delete packages, depending on the `module.vars.state` parameter """
+        if len(packages) == 0:
+            return 0, '', ''
+        command_arg = [x.name for x in packages]
+
+        data = 'N'  # Answer 'No' in case sdkmanager wants us to accept license
```

Comment:
This is what I described in the *Additional Information* section of the pull request message. In addition to that, there is a [question](https://forum.ansible.com/t/answering-prompts-from-ansible-modules/11193) I asked on the forum, which is simplified version of the issue I am facing here

Reply:
Sorry, my bad. My brain let go of that part way too quickly. 

TBH, not sure how to solve that. The only thing I can think of is: if you can have a guarantee that each SDK will have only one license, then you make the `name` parameter a `str` instead of `list[str]` and use `data="N"` (or `"Y"`) in the command run. Naturally, that would create a hit in performance for playbooks with multiple SDKs, because they would need to iterate over the list of SDKs at the playbook level, instead of Python's. Choices, choices.

AFAICT, there is no way to dynamically calculate the data (stdin) of the command in the way it is implemented in Ansible today. That might be an interesting addition to `ansible-core` in the future though.

Reply:
Well, the thing is that google didn't bother to document it anywhere. Some packages might not have any licenses, some do. I'm not aware whether there are packages that belong to multiple licenses at the same time. 
Anyway, even the official recommendation of using Android SDK on build servers is to accept all licenses at once before any packages are installed, so if you think it is acceptable, I'd leave it as is (let the module assume that all packages were accepted in advance)

Reply:
Well, it is what it is and, as you said, google didn't bother so there isn't much we can do.

The concern I have, though, is not so much about the `yes | sdkmanager` part but about this code that sets stdin data to `"N"`. If one particular sdk has more than one licence, then one single "N" will not be enough, and that will lead to that situation where the module hangs on the second prompt.

If multiple Ns is acceptable (will need to be tested) even when `sdkmanager` prompts only once or never at all, like:
```python
        data = 'N\nN\nN\nN'  # Answer 'No' in case sdkmanager wants us to accept license
```
then what prevents us from doing:
```python
        data = 'Y\nY\nY\nY'
```
?

Reply:
Another way - just occurred while writing the last comment - is to create a module util (may in or around `CmdRunner`) that replicates the behaviour of the `yes` command. That would probably need to import directly from `subprocess` rather than using `AnsibleModule.run_command()`. And that would clearly be a different PR.

Reply:
> Well, it is what it is and, as you said, google didn't bother so there isn't much we can do.
> 
> The concern I have, though, is not so much about the `yes | sdkmanager` part but about this code that sets stdin data to `"N"`. If one particular sdk has more than one licence, then one single "N" will not be enough, and that will lead to that situation where the module hangs on the second prompt.
> 
> If multiple Ns is acceptable (will need to be tested) even when `sdkmanager` prompts only once or never at all, like:
> 
> ```python
>         data = 'N\nN\nN\nN'  # Answer 'No' in case sdkmanager wants us to accept license
> ```
> 
> then what prevents us from doing:
> 
> ```python
>         data = 'Y\nY\nY\nY'
> ```
> 
> ?

From what I can see, no packages belong to multiple licenses, so we have at most 1 license per package. However, when packages are installed in bulk (as it is implemented right now), `sdkmanager` will prompt for each not accepted license for packages requested to be installed. 
After some testing it looks like you are right that sending single 'N' won't save the situation, as in case of several prompts it asks second time after the very first prompt is declined (when I was implementing this, I wrongfully assumed that in case of multiple prompts, if at least one prompt is declined, the whole prompt chain gets cancelled). 
Regarding your suggestion about sending 'Y\nY\nY\nY'  -- I have tested this and it seems to be working ok (however, the count of 'Y' should be just a wild guess, since there isn't a documentation saying how many licenses are there.  Also, I suspect that in future google might add new packages to `sdkmanager` and may add new licenses as well, so the string should be long enough to handle that). Additionally, it works even if the amount of 'Y' exceeds the amount of licenses that need to be accepted.
Another possible solution I found might be in installing packages one at a time. I tried installing all distinct package this way and it seems that the assumption that there is at most 1 license per package. The possible implementation:
````python
    def apply_packages_changes(self, packages):
        """ Install or delete packages, depending on the `module.vars.state` parameter """
        if len(packages) == 0:
            return 0, '', ''

        for package in packages:
            data = 'y'
            with self.runner('state name sdk_root channel', data=data) as ctx:
                rc, stdout, stderr = ctx.run(name=package.name, data=data)

                if rc != 0:
                    self._try_parse_stderr(stderr)
                    return rc, stdout, stderr
        return 0, '', ''
````

Reply:
@russoz Also, asking a bit in advance, but if we figure out the correct way to accept licenses automatically, what do you think about adding a module parameter `accept_licenses` (bool), so that the users of the module would have to explicitly set this parameter to `true` to have all the necessary licenses accepted.

Reply:
Ok, unpacking:
* accept_license flag: good idea!
* now about installing packages one at a time: that solution will clearly have a performance impact. If you are feeling brave enough, you might have both implementations (one at a time, all at once) and add another feature flag to control which one is used. 
  * if you take that path, you might want to calculate the number of "N" (or "Y") like: `"\n".join(["N"] * len(name))`, perhaps times 2 to allow for discrepancies.
* regardless of you adopting the suggestion or not, or how you implement it, the docs must explain the risks and the situation to the users. If something changes in the future (like a SDK with 0 or 3 licences, or a new version of sdkmanager that makes one additional question, etc...), users should be able to refer to the module docs and have some understanding about the problem.

There's only so much one can do when the tool (sdkmanager) does not cooperate. I would suggest - if you have not done that yet - to raise a ticket with them to add a parameter like `--accept-all-licenses` or similar.